### ISSUE TITLE
Translate Website into all languages

### ISSUE THREAD

**Issue by MaxHillebrand** [zkSNACKs/WasabiWalletWebSite#3]
English is not enough, the website should be readable in all languages.

1. Set up a transifex.com account
2. Connect this website repository to transifex
3. Get deepl / google translate API key
4. Set up translation CI, each commit to master gets automagically translated
5. Review the translated content in some languages to see if full automation is shippable
6. Ship it

After we figure out an automated process for this website, do the same with the documentation and the software.

**Comment by nopara73** [comment]
This is the start of a big project as it will set the direction for the localization efforts of the software itself. Would it be reasonable to have a person responsible for localizations for the whole project eventually and that this person does an analysis of different kind of approaches before doing anything? Like traditional hiring of translators, crowdsourcing translations, using online translator APIs, building our own custom translations with GPT-4 and whatnot.

**Comment by MarnixCroes** [comment]
For all webpages (main website, docs & blog), it should be fairly easy if we decide to use Google Translate API.
This would have the same translations as if you enter a [URL in google translate](https://translate.google.com/?sl=auto&tl=en&op=websites).
AFAICT these translations are near perfect.

For the website, the UX would look something like this:

[Screencast from 10-08-2023 12:07:55.webm](https://github.com/zkSNACKs/WasabiWalletWebSite/assets/93143998/e25e05a5-b512-4821-bdd8-6779cc58387b)

Possibility to translate into 104 languages.

### Plan
Google doesn't offer this for free.
They have several plans, the [basic plan](https://cloud.google.com/translate/#section-5) would be sufficient for us (HTML text translation).
### Price
The price is per character sent to the API, I think we would fall in the category of _500,000 to 1 billion characters per month_, which is billed as $20 per million characters.


### Considerations
- Some browsers already offer instantaneous translations (I read Chrome can do this?), I suspect that behavior like this will continue to be more widespread, so eventually working on this might be obsolete (not sure what time frame though).

- Not sure about FOSS options, I'd have to check.

- Don't be dependent on someone else's API, but this will require a lot more effort (manual translations?)

**Comment by MaxHillebrand** [comment]
Thanks, do we store the translated text in github? Or is this done "on the fly" every time a user clicks it?

**Comment by MarnixCroes** [comment]
> Or is this done "on the fly" every time a user clicks it?

This

**Comment by kristapsk** [comment]
> AFAICT these translations are near perfect.

Did fast check for Latvian, and it's clearly not. And I looked at it only for few seconds. Same could be for other smaller languages.

"Open-source, non-custodial Bitcoin Wallet for desktop" is translated as "Atvērtā koda, ar brīvības atņemšanu nesaistīts
Bitcoin maks darbvirsmai". Google translates "non-custodial" to "ar brīvības atņemšanu nesaistīts", but that phrase in Latvian is used as "ar brīvības atņemšanu nesaistīts drošības līdzeklis" which translates to "a non-custodial measure of security", which is term used during criminal cases (it means, they don't put you in the jail during investigation, instead there is home arrest or you are forbidden to leave country, etc). :)

**Comment by nopara73** [comment]
Hungarian @CAnorbo

<img width="531" alt="image" src="https://github.com/zkSNACKs/WasabiWalletWebSite/assets/9156103/4713058f-01ff-4663-9d04-659a7555efb0">

<img width="1110" alt="image" src="https://github.com/zkSNACKs/WasabiWalletWebSite/assets/9156103/9b329b6c-7973-49b8-9810-5d64cda05821">

<img width="1190" alt="image" src="https://github.com/zkSNACKs/WasabiWalletWebSite/assets/9156103/8b40bf12-c8e6-456b-99b5-5e3742384fea">

**Comment by CAnorbo** [comment]
LoL Terrible Hungarian translation. 😅

**Comment by kristapsk** [comment]
> LoL Terrible Hungarian translation. 😅

Is it at least something funny?

**Comment by CAnorbo** [comment]
My favourite is "nem örizetbe vett" 🤣🤣

**Comment by nopara73** [comment]
"nulla tudású szoftver" is a pretty dope expression though. I'm going to start using it now :D

**Comment by CAnorbo** [comment]
> My favourite is "nem örizetbe vett" 🤣🤣

This means "not arrested" and the "nulla tudású szoftver" means that this software knows nothing, like no functionality.

**Comment by nopara73** [comment]
Another alternative is to add a thin band on top of the website that says "This website was auto-translated by an AI robot slave to Hungarian. Click here to switch to English" and then wait a decade until the technology catches up. AI auto-translation is clearly the future and we don't want to get stuck in a cave with rocks and stones.

**Comment by CAnorbo** [comment]
Btw, I think this too techical content is not even translatable to our Alien language properly.

**Comment by molnard** [comment]
> This is the start of a big project as it will set the direction for the localization efforts of the software itself. Would it be reasonable to have a person responsible for localizations for the whole project eventually and that this person does an analysis of different kind of approaches before doing anything?

I completely agree with this. And do not forget to figure out the customer support side as well.

**Comment by MarnixCroes** [comment]
>  AFAICT these translations are near perfect.

https://github.com/zkSNACKs/WasabiWalletWebSite/assets/93143998/53a96059-7974-4170-ad50-ac7b0a8ed797

at least for Dutch it's pretty good :D

**Comment by MarnixCroes** [comment]
FWIW, the Google translate API doesn't work for onion service
